inception/inception-api-annotation-experimental/src/main/ts/client/AnnotationExperienceAPIImpl.ts

```diff
@@ -16,293 +16,448 @@
+import {AllSpanResponse} from "./messages/response/span/AllSpanResponse";
+import {AllRelationResponse} from "./messages/response/relation/AllRelationResponse";
+import {AllSpanRequest} from "./messages/request/span/AllSpanRequest";
+import {AllRelationRequest} from "./messages/request/relation/AllRelationRequest";
+import {Viewport} from "./model/Viewport";
 
 export class AnnotationExperienceAPIImpl implements AnnotationExperienceAPI {
 
     //Websocket and stomp broker
-    stompClient: Client;
-    connected: boolean = false;
+    private _stompClient: Client;
 
     //States to remember by client
-    client: string;
-    projectID: string;
-
-    documentID: string;
-
+    private _clientName: string;
+    private _projectID: number;
+    private _documentID: number;
 
     //Text and annotations
-    text: String[];
-    annotations: Annotation[];
-    selectedAnnotation: Annotation;
+    private _text: string[];
+    private _spans: Span[];
+    private _selectedSpan: Span;
 
-    //Viewport
-    viewport: number[][];
+    private _relations: Relation[];
+    private _selectedRelation: Relation;
 
-    //Visualizer
-    readonly visualizer: AnnotationExperienceAPIVisualization;
-
-    //Actionhandler
-    readonly actionhandler: AnnotationExperienceAPIActionHandler;
+    //Viewport
+    private _viewport: Viewport;
 
     constructor() {
         this.connect();
-
-        //Visualizer
-        this.visualizer = new AnnotationExperienceAPIVisualization(this);
-
-        //ActionHandler
-        this.actionhandler = new AnnotationExperienceAPIActionHandler(this);
-
-        this.actionhandler.registerDefaultActionHandler();
     }
 
 
     //CREATE WEBSOCKET CONNECTION
     connect() {
-        if (this.connected) {
-            console.log("You are already connected")
-            return;
-        }
 
-        this.stompClient = Stomp.over(function () {
+        //TODO find better solution
+
+        this._stompClient = Stomp.over(function () {
             return new WebSocket(localStorage.getItem("url"));
         });
 
+        this._viewport = new Viewport([[0,79],[80,140],[141,189]],null);
+
         const that = this;
 
-        this.stompClient.onConnect = function (frame) {
-            that.connected = true;
+        this._stompClient.onConnect = function (frame) {
 
             //Receive username from inital message exchange header
             const header = frame.headers;
             let data: keyof typeof header;
             for (data in header) {
-                that.client = header[data];
+                that._clientName = header[data];
                 break;
             }
 
             //Receive project and document from URL
 
-            that.projectID = document.location.href.split("/")[5];
-            that.documentID = document.location.href.split("=")[1].split("&")[0];
+            that._projectID = Number(document.location.href.split("/")[5]);
```

Comment:
Do you support switching between multiple documents in a project? What happens if I use the next-doc/prev-doc buttons? Do you "listen" to the URL changing?

Reply:
Hm this is actually an issue on how we want to solve the viewport:
Listening to the URL on a click (which will be handled in each editors ActionHandler.class - registerDefaultActionHandler()) is pretty simple, however INCEpTION uses sentences in the URL right (e.g. ..&f=6)? And in the editors settings we can configure how many sentences are shown?
So as you suggested I should not use sentences (or at least not decide on offset / word / sentence) but only offsets.
How should the client know what the offset begin of sentence 6 and the end in sentence 11 (when the pagesize is 5) is with the offset approach?
Maybe use a completly different approach than it was before with arrows for paging and URL change?